machine/v1beta1/types_azureprovider.go

```diff
@@ -174,6 +174,13 @@ type Image struct {
 	Version string `json:"version"`
 	// ResourceID specifies an image to use by ID
 	ResourceID string `json:"resourceID"`
+	// Type identifies the source of the image and related information, such as purchase plans.
+	// Valid values are "ID", "MarketplaceWithPlan", "MarketplaceNoPlan", and omitted, which
+	// behaves as MarketplaceNoPlan if publisher data is supplied, or ID if not.
+	// For more information about purchase plans, see:
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage#check-the-purchase-plan-information
+	// +optional
+	Type string `json:"type,omitempty"`
```

Comment:
I think this works from a description point of view, I would be tempted to add the valid values as constants and make this a typed string, this should then give us constsants to use during the implementation which should make things cleaner

```
type AzureImageType string

const (
  AzureImageTypeID AzureImageType = "ID"
  AzureImageTypeMarketplaceNoPlan AzureImageType = "MarketPlaceNoPlan"
  AzureImageTypeMarketplaceWithPlan AzureImageType = "MarketPlaceWithPlan"
)
```

WDYT?